src/screens/ChatScreen/SignedChatScreen.tsx

```diff
@@ -84,9 +86,29 @@ const SignedChatScreen = () => {
     if (selectedChannel) {
       setChannel(selectedChannel, dispatchChannel);
       fetchOtherProfile();
+      fetchChannelDetail(selectedChannel as ChannelList);
     }
   }, [selectedChannel]);
 
+  const appState = React.useRef(AppState.currentState);
+
+  React.useEffect(() => {
+    const subscription = AppState.addEventListener('change', (nextAppState) => {
+      if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
+        fetchChannelDetail(selectedChannel as ChannelList);
+        if (selectedChannel) {
+          fetchChannelDetail(selectedChannel as ChannelList);
+        }
+      }
+
+      appState.current = nextAppState;
+    });
+
+    return () => {
+      subscription?.remove();
+    };
+  }, [selectedChannel]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There seems to be a redundancy in your code. You are calling `fetchChannelDetail(selectedChannel as ChannelList);` twice when the app state changes from inactive or background to active. This could lead to unnecessary network requests and might slow down the application. Here's how you can fix it:

```diff
  React.useEffect(() => {
    const subscription = AppState.addEventListener('change', (nextAppState) => {
      if (appState.current.match(/inactive|background/) && nextAppState === 'active') {
        if (selectedChannel) {
          fetchChannelDetail(selectedChannel as ChannelList);
        }
      }

      appState.current = nextAppState;
    });

    return () => {
      subscription?.remove();
    };
  }, [selectedChannel]);
```
This way, you only call `fetchChannelDetail` once when the app state changes and `selectedChannel` is not null.


<!-- This is an auto-generated comment by OSS CodeRabbit -->